http://xbmc.org/trac/ticket/5946
http://xbmc.org/forum/archive/index.php/t-29076.html

we cannot call snd_pcm_bytes_to_frames() as it requires the hw_params to be
committed via snd_pcm_hw_params() as that is what sets up pcm->frame_bits

--- XBMC/xbmc/cores/AudioRenderers/ALSADirectSound.cpp
+++ XBMC/xbmc/cores/AudioRenderers/ALSADirectSound.cpp
@@ -200,7 +200,7 @@
   nErr = snd_pcm_hw_params_set_period_size_near(m_pPlayHandle, hw_params, &m_maxFrames, 0);
   CHECK_ALSA_RETURN(LOGERROR,"hw_params_set_period_size",nErr);
 
-  m_BufferSize = snd_pcm_bytes_to_frames(m_pPlayHandle,m_dwPacketSize * 10); // buffer big enough - but not too big...
+  m_BufferSize = m_dwPacketSize * 10; // buffer big enough - but not too big...
   nErr = snd_pcm_hw_params_set_buffer_size_near(m_pPlayHandle, hw_params, &m_BufferSize);
   CHECK_ALSA_RETURN(LOGERROR,"hw_params_set_buffer_size",nErr);
 
